tools/src/wyvern/target/corewyvernIL/support/GenUtil.java

```diff
@@ -57,6 +58,11 @@
 	public static Expression doGenModuleIL(GenContext ctx, GenContext addedCtx, GenContext origCtx, Iterator<? extends TypedAST> ai, boolean isModule) {
 		if (ai.hasNext()) {
 			TypedAST ast = ai.next(); 
+			if (ast instanceof TypeAbbrevDeclaration) {
+				TypeAbbrevDeclaration typeAbbrevDecl = (TypeAbbrevDeclaration) ast;
+				AliasGenContext newCtx = new AliasGenContext(typeAbbrevDecl.getAlias(), typeAbbrevDecl.getReference(), ctx);
+				return doGenModuleIL(newCtx,addedCtx, origCtx, ai, isModule);
+			} else 
 			// TODO (BUG): the approach of wrapping at the end won't work for VarDeclarations!
```

Comment:
Here, TypeAbbrevDeclaration should work just like TypeVarDecl (and DefDeclaration) instead of a special case (and AliasGenContext).  Just add another disjunction to the if statement for them and implement the right methods on TypeAbbrevDeclaration, notably topLevelGen() and genILType().  You'll also need to extend GenContext.rec() to be aware of TypeAbbrevDeclaration (actually we should refactor rec to be a method on decls, not on GenContext, but we can fix that separately).